taskchampion/src/storage/inmemory.rs

```diff
@@ -49,6 +49,24 @@ impl<'t> StorageTxn for Txn<'t> {
         }
     }
 
+    fn get_pending_tasks(&mut self) -> Result<Vec<(Uuid, TaskMap)>> {
+        let res = self
+            .get_working_set()?
+            .iter()
+            .filter_map(|uuid| {
+                uuid.map(|inner_uuid| {
+                    self.data_ref()
+                        .tasks
+                        .get(&inner_uuid)
+                        .map(|taskmap| (inner_uuid, taskmap.clone()))
+                })
+                .flatten()
+            })
+            .collect::<Vec<_>>();
```

Comment:
This is nice! It took me a bit to realize that `.get(..)` also returned an option, resulting in `Option<Option<(uuid, TaskMap)>>` and requiring the `.flatten()`.

Reply:
This is the part that I dislike because it's not readable enough

I'm not smart enough to figure out how to make it cleaner, so I've added a comment about what is going on to make it a bit more clear

Reply:
That's probably good. I think this is actually pretty idiomatic Rust, but it does require a lot of context about the various types involved to understand it. The comment addresses that perfectly. 💯